forum/profiles/tests.py

```diff
@@ -1,3 +1,215 @@
+
+    def test_create_investor_profile_invalid_balance_bigger(self):
+        url = reverse("profiles:investor-profile-list")
+        response = self.client.post(
+            url, self.invalid_balance_data_bigger_balance, format="json"
+        )
+        print(response.data)
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertIn("account_balance", response.data)
+
+    def test_create_investor_profile_invalid_balance_smaller(self):
+        url = reverse("profiles:investor-profile-list")
+        response = self.client.post(
+            url, self.invalid_balance_data_smaller_balance, format="json"
+        )
+        print(response.data)
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertIn("account_balance", response.data)
+
+    def test_update_investor_profile(self):
+        url = reverse("profiles:investor-profile-list")
+        response = self.client.post(url, self.data, format="json")
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+        investor_profile_id = response.data["id"]
+
+        update_url = reverse(
+            "profiles:investor-profile-detail", args=[investor_profile_id]
+        )
+        updated_data = self.data.copy()
+        updated_data["city"] = "Kyiv"
+        response = self.client.put(update_url, updated_data, format="json")
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.data["city"], "Kyiv")
+
+    def test_partial_update_investor_profile(self):
+
+        url = reverse("profiles:investor-profile-list")
+        response = self.client.post(url, self.data, format="json")
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+        investor_profile_id = response.data["id"]
+
+        partial_update_url = reverse(
+            "profiles:investor-profile-detail", args=[investor_profile_id]
+        )
+        partial_update_data = {"city": "Odessa"}
+        response = self.client.patch(
+            partial_update_url, partial_update_data, format="json"
+        )
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.data["city"], "Odessa")
+
+    def test_delete_investor_profile(self):
+        url = reverse("profiles:investor-profile-list")
+        response = self.client.post(url, self.data, format="json")
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+        investor_profile_id = response.data["id"]
+
+        delete_url = reverse(
+            "profiles:investor-profile-detail", args=[investor_profile_id]
+        )
+        response = self.client.delete(delete_url)
+        self.assertEqual(response.status_code, status.HTTP_204_NO_CONTENT)
+
+        response = self.client.get(delete_url)
+        self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
+
+    def tearDown(self):
+        del self.user
+        del self.token
```

Comment:
The tearDown method is not strictly necessary in Django tests, as Django’s TestCase class automatically manages a mock database. After running all tests, Django cleans up the test database to ensure that no changes persist between test runs. However, I added this method based on your request, as it can help to clarify the cleanup process or manage additional resources.